Add documentation on how to handle moving meshes (#381)
precice/precice.github.io: pages/docs/couple-your-code/couple-your-code-moving-meshes.md

title: Dealing with moving meshes
permalink: couple-your-code-moving-meshes.html
keywords: api, adapter, ALE, geometries, CFD
summary: "preCICE supports ALE-methods, direct-mesh-access, and pseudo-meshes to handle most scenarios of moving meshes. A remeshing API is on the roadmap and being actively developped."
---

## Static reference domain

preCICE supports static meshes, meaning dynamic meshes can still be emulated using node displacements as described by the Arbitrary-Lagrangian-Eulerian or ALE-methods.

> Biggest downside of this method is that spatially aware mapping-schemes in preCICE work in the reference domain, not the physical domain.

Reviewer: I would not necessarily call this a downside.

Author: ```suggestion
Note that spatially aware mapping-schemes in preCICE work in the reference domain, not the physical domain.
```

Examples of this method are the [perpendicular flap tutorials](tutorials-perpendicular-flap) and the [Turek-Hron FSI3 benchmark](tutorials-turek-hron-fsi3).

## Mapping inside adapter

> In some cases, the mesh of a participant may fluctuate extremely between timestep, common for particle-based methods.

Reviewer: I guess you need to explain what you mean with "mesh for particle-based methods" here.

Author: ```suggestion
In some cases, the mesh of a participant may fluctuate extremely between timestep, common for Discrete Element Methods.
```

Reviewer: what I meant is to explain why there is a mesh for a particle method. meaning that we consider the particle locations as mesh locations. this is not obvious, but a preCICE thing.

Author: Makes sense. I reworded the intro and also added overset meshes, as they may fall into the same case.
